Source/Dafny/Compiler-java.cs

```diff
@@ -1978,7 +1978,9 @@ private class GenericArrayElementLvalue : ILvalue {
 
     protected void EmitToString(TargetWriter wr, Expression arg) {
       if (arg.Type.IsArrowType) {
-        wr.Write(IdName(((IdentifierExpr) ((ConcreteSyntaxExpression)arg).ResolvedExpression).Var) + " == null ? null : \"Function\"");
+        var expr = ((ConcreteSyntaxExpression)arg).ResolvedExpression;
+        if (expr is IdentifierExpr) wr.Write(IdName(((IdentifierExpr) expr).Var) + " == null ? null : \"Function\"");
+        else wr.Write("\"Function\"");
```

Comment:
Always use curly braces around both "then" and `else` alternatives.

Also, instead of writing `expr is IdentifierExpr` and then doing the cast, C# allows you to write

    if (expr is IdentifierExpr x) {
        ...  x.Var ...

The mechanism is not perfect, because of where `x` is scoped, but it works well in this case.

Reply:
handy if hacky